File: src/main/java/restaurantmanager/employee/Employee.java
package restaurantmanager.employee;

import java.math.BigDecimal;
import java.time.LocalDate;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@Builder(toBuilder = true)
@AllArgsConstructor(access = AccessLevel.PACKAGE)
@NoArgsConstructor(access = AccessLevel.PUBLIC)
@Getter
@ToString
@EqualsAndHashCode
@Table(name = "employees")
public class Employee {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String firstName;
	
	private String lastName;
	
	private String email;
	
	private String position;
	
	private BigDecimal salary;
	
	@Column(name = "phone_number")
	private String phoneNumber;
	
	@Column(name = "start_date")
	private LocalDate startDate;
	
}

File: src/main/kotlin/restaurant_manager/infrastructure/Employee.kt
package restaurant_manager.infrastructure

import org.hibernate.annotations.ColumnDefault
import restaurant_manager.enums.Position
import restaurant_manager.others.PhoneNumber
import restaurant_manager.others.Salary
import java.time.LocalDate
import javax.persistence.*

@Entity
@Table(name = "employees")
class Employee(

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,

        var firstName: String,

        var lastName: String,

        var email: String,

        @Enumerated(EnumType.STRING)
        var position: Position,

        @Embedded
        var salary: Salary,

        @ColumnDefault("now()::timestamp::date")
        var startDate: LocalDate,

        var phoneNumber: PhoneNumber

)
